render/data/ch_response.go

```diff
@@ -135,6 +123,43 @@ func (c *CHResponse) ToMultiFetchResponseV3() (*v3pb.MultiFetchResponse, error)
 			}
 			mfr.Metrics = append(mfr.Metrics, fr)
 		}
+		return nil
+	}
+
+	// process metrics with points
+	writtenMetrics := make(map[string]struct{})
+	nextMetric := data.GroupByMetric()
+	for {
+		points := nextMetric()
+		if len(points) == 0 {
+			break
+		}
+		id := points[0].MetricID
+		name := data.MetricName(id)
+		writtenMetrics[name] = struct{}{}
+		consolidationFunc, err := data.GetAggregation(id)
+		if err != nil {
+			return nil, err
+		}
+		step, err := data.GetStep(id)
+		if err != nil {
+			return nil, err
+		}
+		if err := addResponse(name, consolidationFunc, uint32(c.From), uint32(c.Until), step, points); err != nil {
+			return nil, err
+		}
+	}
+	// process metrics with no points
+	if c.AppendOutEmptySeries && len(writtenMetrics) < data.AM.Len() && data.CommonStep > 0 {
+		for _, metricName := range data.AM.Series(false) {
+			if _, done := writtenMetrics[metricName]; done {
```

Comment:
May be simplify to
`if _, done := writtenMetrics[metricName]; !done {`
and avoid of continue block ?